internal/resource.go

```diff
@@ -63,13 +63,32 @@ func getUrl(u string, fileName string, ctx context.Context) (string, error) {
 	return fileName, nil
 }
 
-// GetUrlToDir downloads the given resource to the given directory and returns the path to it.
-func GetUrlToDir(u string, targetDir string, ctx context.Context) (string, error) {
+// GetUrlToDir downloads the given resource to a temporary file and returns the path to it.
+// Modify the GetUrlToDir function to ensure proper file cleanup
+func GetUrlToDir(u string, targetDir string, ctx context.Context) 
+(string, error) {
 	// create temporary name in the target directory.
 	h := sha256.New()
 	h.Write([]byte(u))
 	fileName := filepath.Join(targetDir, fmt.Sprintf(".%s", hex.EncodeToString(h.Sum(nil))))
-	return getUrl(u, fileName, ctx)
+
+	// Remove existing file if present
+	os.Remove(fileName)
+
+	// Create new file with immediate close
```

Comment:
Why the need to do this?

Reply:
You've marked this resolved but you have not answered my question.

Reply:
| this is not included and has been deleted in the final 1 commit apologies for not communicating that with you raphael, last week in the last few commits i did it was fixed

Do you mean that this code was removed? I see it's still very much here.

Reply:
Code is still here...